File: backend/src/main/java/com/hotelbooking/backend/repositories/BookingRepository.java
package com.hotelbooking.backend.repositories;

import com.hotelbooking.backend.domain.entities.BookingEntity;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDate;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface BookingRepository extends JpaRepository<BookingEntity, UUID> {

    List<BookingEntity> findByRoomIdAndStatusInAndCheckInBeforeAndCheckOutAfter(
            UUID roomId,
            List<Integer> statuses,
            LocalDate checkIn,
            LocalDate checkOut
    );

    List<BookingEntity> findByUserIdAndStatusInOrderByCheckInAsc(
            UUID userId,
            List<Integer> statuses
    );

    List<BookingEntity> findByUserId(UUID userID);

    Optional<BookingEntity> findById(UUID bookingCode);

    List<BookingEntity> findByRoomRoomTypeHotelId(UUID hotelId);
}

File: mobile/app/src/main/java/com/hotelbooking/mobile/domain/repository/BookingRepository.kt
package com.hotelbooking.mobile.domain.repository

import com.hotelbooking.mobile.domain.model.Booking
import java.time.LocalDate
import java.util.UUID

interface BookingRepository {
    suspend fun checkAvailability(roomId: UUID, checkIn: LocalDate, checkOut: LocalDate): Result<Boolean>
    suspend fun createBooking(roomId: UUID, checkIn: LocalDate, checkOut: LocalDate): Result<Booking>
    suspend fun getMyBookings(): Result<List<Booking>>
    suspend fun cancelBooking(bookingId: UUID): Result<Unit>
}
